File: PersonalPortfolioTrackerBackend/PersonalPortfolioTracker/DTOs/Request/AddTransactionRequestDto.cs
using System.Text.Json.Serialization;

namespace PersonalPortfolioTracker.DTOs.Request
{
    public class AddTransactionRequestDto
    {
        [JsonPropertyName("stockCode")]
        public string? StockCode { get; set; }

        [JsonPropertyName("date")]
        public string? Date { get; set; }

        [JsonPropertyName("type")]
        public string? Type { get; set; }

        [JsonPropertyName("noOfSharesTransacted")]
        public int NoOfSharesTransacted { get; set; }

        [JsonPropertyName("transactedPricePerShare")]
        public decimal TransactedPricePerShare { get; set; }

        [JsonPropertyName("totalTransactionRelatedExpenses")]
        public decimal TotalTransactionRelatedExpenses { get; set; }
    }
}

File: personal-portfolio-tracker-frontend/src/dtos/Request/AddTransactionRequestDto.js
export default class AddTransactionRequestDto {
  constructor(stockCode, date, type, noOfSharesTransacted, transactedPricePerShare, totalTransactionRelatedExpenses) {
    this.stockCode = stockCode;
    this.date = date;
    this.type = type;
    this.noOfSharesTransacted = noOfSharesTransacted;
    this.transactedPricePerShare = transactedPricePerShare;
    this.totalTransactionRelatedExpenses = totalTransactionRelatedExpenses;
  }
};